pos_container_deposit/models/product_template.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2024 Hunki Enterprises BV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    is_deposit = fields.Boolean(
+        help="Check this if this product is a container for which you charge deposit",
+    )
+    deposit_product_id = fields.Many2one(
+        "product.product",
+        "Deposit product",
+        domain=[("is_deposit", "!=", False)],
+        help="If this product is packaged in a container for which you charge deposit, "
+        "add a product here that stands for the deposit",
+        compute="_compute_deposit_product_id",
+        inverse="_inverse_deposit_product_id",
+        search="_search_deposit_product_id",
+    )
+
+    @api.depends("product_variant_ids.deposit_product_id")
+    def _compute_deposit_product_id(self):
+        for this in self:
+            this.deposit_product_id = this.product_variant_ids.deposit_product_id
```

Comment:
AFAICS, there's nothing to prevent variants to have different deposit products.
Then, this line here would fail trying to assign a recordset with len > 1, won't it?

Perhaps you could align with the mechanism that in place in standard for the weight, volume, etc..
That is:
* If it's a single variant product, read/set the field on the variant
* Otherwise, prevent writing and read `False` + hide the field in the view